integration-tests/tests/src/tests/sync/mongo-db-client.ts

```diff
@@ -0,0 +1,188 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2023 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+import { expect } from "chai";
+import { ChangeEvent, Document, InsertEvent, MongoDBCollection } from "realm";
+
+import { authenticateUserBefore, importAppBefore } from "../../hooks";
+import { sleep } from "../../utils/sleep";
+
+type TestDocument = {
+  _id: number;
+  text?: string;
+  isLast?: boolean;
+};
+
+describe.skipIf(environment.missingServer, "MongoDB Client", function () {
+  this.timeout(60_000); // TODO: Temporarily hardcoded until envs are set up.
+  importAppBefore("with-db");
+  authenticateUserBefore();
+
+  const serviceName = "mongo-db";
+  const dbName = "test-database";
```

Comment:
Yes the implemented tests all pass. Compared to the lines of code you linked to, the `importAppBefore()` on [bindgen](https://github.com/realm/realm-js/blob/a8ef5b587ff1c8628363c216e4932997c50f5d78/integration-tests/tests/src/hooks/import-app-before.ts#L36) use:
```ts
this.app = await importApp(name, replacements);
```

Reply:
Inside the `importApp` function, you will see some logic for [generating a database name](https://github.com/realm/realm-js/blob/test-migration-master/integration-tests/tests/src/utils/import-app.ts#L99).  I did some refactoring to this code in the `test-migration-master` so that the generated db-name was available in `this.database`.  We should probably get those upstreamed sooner than later.

Reply:
Yeah that'd be great!